tracer/src/Datadog.Trace/Debugger/Instrumentation/AsyncLineDebuggerInvoker.cs

```diff
@@ -137,25 +140,32 @@ public static AsyncLineDebuggerState BeginLine<TTarget>(string probeId, TTarget
                         Log.Warning($"{nameof(BeginLine)}: hoisted 'this' has not found. {kickoffInfo.KickoffParentType.Name}.{kickoffInfo.KickoffMethod.Name}");
                     }
 
-                    if (!MethodMetadataProvider.TryCreateAsyncMethodMetadataIfNotExists(instance, methodMetadataIndex, in methodHandle, in typeHandle, kickoffInfo))
+                    if (!MethodMetadataRegistry.Instance.TryCreateAsyncMethodMetadataIfNotExists(instance, methodMetadataIndex, in methodHandle, in typeHandle, kickoffInfo))
                     {
-                        Log.Warning($"BeginMethod_StartMarker: Failed to receive the InstrumentedMethodInfo associated with the executing method. type = {typeof(TTarget)}, instance type name = {instance.GetType().Name}, methodMetadataId = {methodMetadataIndex}");
+                        Log.Warning($"([Async]BeginLine: Failed to receive the InstrumentedMethodInfo associated with the executing method. type = {typeof(TTarget)}, instance type name = {instance.GetType().Name}, methodMetadataId = {methodMetadataIndex}, probeId = {probeId}");
```

Comment:
I know it wasn't introduced in this PR, but we should fix all the cases in this file where we send Serilog an interpolated string instead of just a constant template, as Serilog caches these strings and it would cause a memory usage hit and performance degradation. Please see "Templates vs. Messages" in https://benfoster.io/blog/serilog-best-practices/#message-template-recommendations

Reply:
There are plenty of places where we use string concat instead of a constant template and since the point in the code you are referring is an edge case we will rarely ever encounter in production, I would say let's defer this change to it's own PR pre-GA where we will replace every single string interpolation in Serilog with a constant message. WDYT?

Reply:
Just to point out we similarly need to switch to message templates instead of concat ASAP, as log messages will rely on this for redaction. Essentially we will record the log _template_ instead of the interpolated log message. If we use string concat, that could lead to exposing information that we don't want to ([that PR isn't merged yet](https://github.com/DataDog/dd-trace-dotnet/pull/3751), so this won't directly impact it yet).

FWIW, I'm going to write an analyzer that automatically checks for these cases and provides code fixes 🙂

Reply:
@andrewlock That's awesome! In that case, I guess it's best if we just wait for your analyzer, as it'd be the most comprehensive and productive way to deal with this. WDYT?